### DISCUSSION TITLE
Publishing values for send-only DPs on Tuya devices and custom parsing/messages

### DISCUSSION THREAD

**Discussion by nikzori** [Koenkk/zigbee2mqtt#23178] [Devices]
Our company is using a send-only DP (boolean in this case) in our water valve controller that we'd like to be able to toggle from z2mqtt frontend. Is this possible to do with an external .js converter? Mapping it doesn't seem to work and makes nodejs throw an error.

Another thing I'm interested in is whether it is possible to display data from a single DP with several exposed properties and compose custom messages to send back to the device. We pack a lot of data in several DPs and use binary masks to get the flags we need, e.g. a DP for a wireless sensor uses a Number type DP (which is just 4-byte integer) with 1 byte reserved for signal strength, 1 for battery level and 2 for various flags. So it would be nice to have several switches and some read-only numeric values instead of a raw integer.

**Comment by Koenkk** [comment]
> Is this possible to do with an external .js converter? Mapping it doesn't seem to work and makes nodejs throw an error.

Should be the same as a send and receive dp, what's the error?

> Another thing I'm interested in is whether it is possible to display data from a single DP with several exposed properties and compose custom messages to send back to the device.

Yes, by making a custom value converter, [example](https://github.com/Koenkk/zigbee-herdsman-converters/blob/d54c80455f12d82851004726c338036cc9397326/src/lib/tuya.ts#L628)

**Reply by nikzori** [reply]
> Should be the same as a send and receive dp

Ah, yep, it is. Forgot to define that specific DP and it kept telling me there is no converter.
Will look into value converters, thanks!
